src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -307,7 +307,12 @@ public bool AllFrameworkVersions
         /// </summary>
         public bool IsUnsupported
         {
-            get { return UnsupportedFramework.Equals(this); }
+            // Poposal 1: the comment above says "true if this framework is invalid or unknown". This makes that happen
+            get
+            {
+                string _;
+                return DefaultFrameworkNameProvider.Instance.TryGetShortIdentifier(_frameworkIdentifier, out _);
```

Comment:
Since the parsed identifier isn't needed, you can use the newer C# feature of "out string _" inline.

Reply:
that's a C#7 feature, and for reasons I don't understand, the NuGet.Frameworks project is locked to C#5

https://github.com/NuGet/NuGet.Client/blob/d421f61d0b180a4878d4d4b0ba1f20c95e8d7817/src/NuGet.Core/NuGet.Frameworks/NuGet.Frameworks.csproj#L10

We should probably get rid of it. I'm not aware of any breaking changes in the language, so I don't see any good reason to lock to an old version.

Thanks for the feedback :)

Reply:
frameworks builds for net40. 
Not sure if it's still necessary.

Reply:
It was added in this commit: https://github.com/NuGet/NuGet.Client/commit/2de2d991f371df898fdd97231883e187f6e2e9bf
which was related to this PR: https://github.com/NuGet/NuGet.Client/pull/587

Given it appears to be added so `NuGet.Framework` can be used by NuGet 2.x (`NuGet.Core`), and we no longer support NuGet 2.x, I think we can remove it.

Reply:
That sounds good. 
We can remove it when we do the framework cleanup. (I want to do it after heng does her changes!)